qutip/tests/solver/test_parallel.py

```diff
@@ -0,0 +1,108 @@
+# This file is part of QuTiP: Quantum Toolbox in Python.
+#
+#    Copyright (c) 2011 and later, Paul D. Nation and Robert J. Johansson.
+#    All rights reserved.
+#
+#    Redistribution and use in source and binary forms, with or without
+#    modification, are permitted provided that the following conditions are
+#    met:
+#
+#    1. Redistributions of source code must retain the above copyright notice,
+#       this list of conditions and the following disclaimer.
+#
+#    2. Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#    3. Neither the name of the QuTiP: Quantum Toolbox in Python nor the names
+#       of its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+#    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+import numpy as np
+import time
+import pytest
+
+from qutip.solver.parallel import parallel_map, serial_map, loky_pmap
+"""
+try:
+    import loky
+except ModuleNotFoundError:
+    loky = False
+"""
+
+def _func1(x):
+    return x**2
+
+
+def _func2(x, a, b, c, d=0, e=0, f=0):
+    assert d > 0
+    assert e > 0
+    assert f > 0
+    time.sleep(np.random.rand() * 0.1)  # random delay
+    return x**2
+
+@pytest.mark.parametrize('map',
+                         [parallel_map, loky_pmap, serial_map],
+                         ids=['parallel_map', 'loky_pmap', 'serial_map'])
```

Comment:
We should use the same form of calling `parametrize` for `test_map` and `test_map_accumulator`. No point in having different styles in the same file.

Reply:
Personally I like the `pytest.param` style best because it keeps all the information line-local, though in this case I _think_ it's also possible to pass an id function `ids=lambda f: f.__name__`